src/analysisd/decoders/dbsync.c

```diff
@@ -173,6 +187,13 @@ static void dispatch_state(dbsync_context_t * ctx) {
     os_malloc(OS_MAXSTR, query);
     os_malloc(OS_MAXSTR, response);
 
+    ComponentType component_type = validate_component(ctx->component);
```

Comment:
We can remove this and centralize the component check at `DispatchDBSync()`

Reply:
Moved!